native/src/modules/layout/containers/HeaderContainer.js

```diff
@@ -53,14 +50,37 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
     state.cityContent.poisRouteMapping[routeKey]
     : null
 
+  if (([OFFERS_ROUTE, DISCLAIMER_ROUTE, SPRUNGBRETT_OFFER_ROUTE]).includes(ownProps.scene.route.name) || shareUrlFromScene) {
```

Comment:
Why do you handle this partly here and partly below? It's a little bit confusing this way

Reply:
you are completly right it is a bit confusing. I did it because afterwards by checking if (!route ...) (line 85) it would exit for every route not being in an routeMapping without setting the route correctly.